instrumentation/github.com/Shopify/sarama/otelsarama/producer.go

```diff
@@ -245,6 +247,31 @@ func WrapAsyncProducer(saramaConfig *sarama.Config, p sarama.AsyncProducer, opts
 	return wrapped
 }
 
+func msgPayloadSize(msg *sarama.ProducerMessage, kafkaVersion sarama.KafkaVersion) int {
```

Comment:
There are a lot of magic numbers in here and it's not clear why/when `binary.MaxVarintLen32` or `binary.MaxVarintLen64` are used.  Some comments would be helpful.